reliability/single-fit/okhttp/src/test/kotlin/com/datadog/android/okhttp/ApolloIntegrationTest.kt

```diff
@@ -0,0 +1,385 @@
+
+    @StringForgery
+    lateinit var fakeResponseBody: String
+
+    @StringForgery
+    lateinit var fakeViewKey: String
+
+    @StringForgery
+    lateinit var fakeViewName: String
+
+    private lateinit var stubSdkCore: StubSDKCore
+    private lateinit var mockServer: MockWebServer
+    private lateinit var apolloClient: ApolloClient
+    private lateinit var rumMonitor: RumMonitor
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        stubSdkCore = StubSDKCore(forge)
+        val registry: Any = Datadog::class.java.getStaticValue("registry")
+        val instances: MutableMap<String, SdkCore> = registry.getFieldValue("instances")
+        instances += stubSdkCore.name to stubSdkCore
+        mockServer = MockWebServer()
+
+        val fakeApplicationId = forge.anAlphabeticalString()
+        val rumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .build()
+        Rum.enable(rumConfiguration, stubSdkCore)
+
+        val fakeTraceConfiguration = TraceConfiguration.Builder().build()
+        Trace.enable(fakeTraceConfiguration, stubSdkCore)
+
+        val tracingInterceptor = createTracingInterceptor()
+        val datadogInterceptor = createDatadogInterceptor()
+        val okHttpClient = OkHttpClient.Builder()
+            .addInterceptor(datadogInterceptor)
+            .addInterceptor(tracingInterceptor)
+            .build()
+
+        apolloClient = ApolloClient
+            .Builder()
+            .serverUrl(mockServer.url("/").toString())
+            .okHttpClient(okHttpClient)
+            .addInterceptor(DatadogApolloInterceptor(sendGraphQLPayloads = true))
+            .build()
+
+        mockServer.enqueue(MockResponse().setResponseCode(200).setBody(fakeResponseBody))
+        rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+    }
+
+    private fun createDatadogInterceptor(): DatadogInterceptor {
+        return DatadogInterceptor
+            .Builder(tracedHosts = listOf(mockServer.hostName))
+            .setSdkInstanceName(stubSdkCore.name)
+            .build()
+    }
+
+    private fun createTracingInterceptor(): TracingInterceptor {
+        return TracingInterceptor
+            .Builder(tracedHosts = listOf(mockServer.hostName)).build()
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        GlobalDatadogTracer.clear()
+        Datadog.stopInstance(stubSdkCore.name)
+        mockServer.shutdown()
+    }
+
+    // region graphQL headers
+
+    @Test
+    fun `M remove GraphQL headers W DatadogInterceptor { with query and Apollo headers }`() = runBlocking {
+        // When
+        apolloClient.query(
+            FakeQuery(
+                userId = fakeUserId,
+                filters = Optional.present(listOf(fakeFilter1, fakeFilter2))
+            )
+        ).execute()
+
+        // Then
+        val requestSent = mockServer.takeRequest()
+        assertThat(requestSent.getHeader(GraphQLHeaders.DD_GRAPHQL_NAME_HEADER.headerValue)).isNull()
```

Comment:
minor: you can simply do `GraphQLHeaders.values().forEach { assertThat(it.headerValue).isNull() }`, this way you won't need to update this code if new values are added